packages/iris-grid/src/IrisGridModel.ts

```diff
@@ -142,6 +145,113 @@ abstract class IrisGridModel<
     return [];
   }
 
+  /**
+   * Parses the column header groups provided.
+   * If undefined, should provide default groups such as from layoutHints
+   *
+   * @returns Object containing groups array, max depth, map of name to parent group, and map of name to group
+   */
+  parseColumnHeaderGroups = memoize(
```

Comment:
Pull this into `IrisGridUtils` as a utility function. Memoization shouldn't be needed in `IrisGridUtils` or `IrisGridTableModelTemplate`; `VisibilityOrderingBuilder` should be calling something like `get initialColumnHeaderGroups` instead.